[IMP] repairs: add services tab (#19483)
odoo/documentation: content/applications/inventory_and_mrp/repairs/repair_orders.rst

Reviewer: ```suggestion
#. :ref:`Process a return order for the damaged product. <repairs/repair_orders/return-order>`
#. :ref:`Create a repair order for the returned product. <repairs/repair_orders/repair>`
#. :ref:`Return the repaired product to the customer .<repairs/repair_orders/return-customer>`
```
I think it's nice to have them be full sentences, since they are so clos to that

Reviewer: ```suggestion
To create a return, navigate to the :menuselection:`Sales app` and click the |SO| containing the products being returned. Click the :guilabel:`Delivery` smart button, which opens the delivery order (DO) form.
```

Reviewer: I don't get a pop-up window at all - I get what appears to be a transfer page. But definitely not a pop-up. Not sure what to call it, so I would rework this so it is accurate this is all outdated according to Runbot. 

<img width="1415" height="725" alt="Image" src="https://github.com/user-attachments/assets/9f291d7c-d926-4e24-aae9-7dec932f36f4" />

Reviewer: I tried in different runbots, so I saw in 19.2 there was a pop-up, but it was called 'Return'. Just verify this information, since I could not!

Reviewer: ```suggestion
To create a new |RO|, navigate to :menuselection:`Repairs app`, and click :guilabel:`New`. This
opens a blank *Repair Orders* form.
```
The name is plural on the form

Reviewer: This image is so small and hard to read, I recommend retaking this and changing the width to 933. Also, this is a small thing I do - but to make the data look better, I change the company name to Stealthy Wood (because I personally hate "My Company' =D )

Reviewer: Also, this can be moved to underneath where all the elemets to fill out are listed,

Reviewer: ```suggestion
On this form, begin by selecting the :guilabel:`Customer` requesting the repairs
```
It sounds odd to say 'invoiced and delivered'. Also, is there a scenario where you'd select a cusotmer to pay but a different return address or cusotmer to ship to? If so, you can add an an admonition block saying 'Tyrpiclaly, the same customer returns the product and pays for the repairs. In cases where a different company pays for the repairs..."

But only if you think that is a valid situation!

Reviewer: ```suggestion
Next, click the :guilabel:`Under Warranty` checkbox, if the product being repaired is covered by a
```

Reviewer: ```suggestion
warranty. If clicked, the :guilabel:`Customer` is not charged for all the parts used in the repair
```

Reviewer: Do they pay labor? Wouldn't the entire repair be free? From the way it is written, it sounds like some parts are free, but the rest may be charged.

Reviewer: Because you have long-form instructions, then a bullet list, I recommend changing this to all bullets. You cna do what I typically do - say:

Fill out the following fields on the form:
- item 1
-  item 2
- item 3

This way, it's more cohesive - because it feels odd to give information is 2 different ways

Reviewer: ```suggestion
The *Parts* tab allows users to specify parts to add, remove, or recycle during the
```
 all tabs are italicized

Reviewer: ```suggestion
The *Repair Notes* tab is used to document important information about the repair. Notes should be added whenever information needs to be recorded for reference, clarification, or documentation.

Common situations where repair notes are necessary include:

- Documenting visible damage or other conditions that may show the repair is not covered under warranty.
- Recording any specific customer requests or instructions that need to be followed during the repair.

```
Just wanted to make this a bit longer!

Reviewer: ```suggestion
Repair order
==========
```
The first one is just 'Return order' to this matches

Reviewer: This image is also hard to read since it's so short, width-wise.

Reviewer: ```suggestion
Click :guilabel:`Return` at the top of the form. This opens a *Reverse Transfer* pop-up
window.
```

Reviewer: I know you didnd't take this imageoriginally, but we shouldn'thave believeable address information on display.

Reviewer: ```suggestion
Click the :icon:`fa-trash-o` :guilabel:`(trash)` icon at the far-right of the product line to remove it from the
```

=============
Repair orders

> =============

Reviewer 2: No meta description? :```(

.. |SO| replace:: :abbr:`SO (Sales Order)`
.. |DO| replace:: :abbr:`DO (Delivery Order)`
.. |RO| replace:: :abbr:`RO (Repair Order)`

Sometimes, products delivered to customers can break or be damaged in transit, and need to be
returned for a refund, delivery of a replacement product, or repairs.

In Odoo, repairs for products returned by customers can be tracked in the **Repairs** app. Once
repaired, products can be redelivered to the customer.

The repair process for damaged products typically follows the below steps:

#. :ref:`Create a repair order <repairs/repair_orders/repair>` for the returned product.
#. :ref:`Process the repair <repairs/repair_orders/process-repair>` for the |RO|.
#. :ref:`Return the repaired product <repairs/repair_orders/return-to-customer>` to the customer.

.. seealso::
   :doc:`../../sales/sales/products_prices/returns`

> .. _repairs/repair_orders/repair:

Reviewer 2: Ref name feels a little generic, but I might just be being fussy.

Create a repair order
=====================

When a damaged product is returned, its repair can be tracked by creating a repair order (RO). To

> create a new |RO|, navigate to the *Repairs app* and click :guilabel:`New`, then fill in the

Reviewer 2: ```suggestion
create a new |RO|, navigate to the **Repairs** app and click :guilabel:`New`, then fill in the
```

   :guilabel:`End Repair`, and a *Consumption Warning* pop-up window appears, listing the expected
   (:guilabel:`To Consume`) and actual (:guilabel:`Consumed`) quantity for each affected product.
   Select one of three options to continue:

   - :guilabel:`Confirm`: Validate the |RO| with the quantities as they currently are.
   - :guilabel:`Set Quantities & Validate`: Update the demand of each line to match the quantity
     consumed before validating.
   - :guilabel:`Discard`: Close the pop-up window without ending the repair.

   .. image:: repair_orders/repair-order-consumption-warning.png
      :alt: Consumption Warning wizard comparing expected and consumed quantities.

Ending the repair moves the |RO| to the :guilabel:`Repaired` stage. If the repair recorded any
product moves, a :guilabel:`Product Moves` smart button also appears above the form. This smart
button does not appear if the |RO| is canceled.

Click the :guilabel:`Product Moves` smart button to view the product's :guilabel:`Inventory Moves`
history during and after the repair process.

.. _repairs/repair_orders/return-to-customer:

Return a product to the customer
================================

> To return a repaired product to the customer, navigate to the :menuselection:`Sales app`, and select

Reviewer 2: I've lost the thread. Should this be **Sales** app, or is **Sales app** correct?

  be manually changed, if needed.

> - :guilabel:`Unit`: Select the |UoM| for the part.

Reviewer: It's odd to start with the Type column, not the product. I would redo this section, and have the fields explained in order. You cna do bullet lists again for this section, or just go in the order they appear. Product, the Type, the Deand.. etc.

> .. image:: repair_orders/repair-orders-parts-tab.png

Reviewer: This image is so small and ahrd to read - I woudl redo this as well, and make the image 933 wide. I woudl also add more parts ot use - have one be recycled, one add, and one remove. You want to show all the things you explained above, if possible, and I think you can.

   :alt: Parts tab on the repair order form with the Quantity column visible.

.. tip::
   To add additional columns to the line, click the :icon:`oi-settings-adjust` :guilabel:`(optional

>    columns drop-down)` icon in the header row. Select the desired options to add to the line.

Reviewer: I feel like you can mention the Deadline field and the SN field - because those are common enough, I would think.

Services tab
~~~~~~~~~~~~

> The :guilabel:`Services` tab allows users to add service-type products, such as labor or

Reviewer: ```suggestion
The *Services* tab allows users to add service-type products, such as labor or
```

diagnostics, to the |RO| for traceability and invoicing purposes. To add a service, click
:guilabel:`Add a line`, then select a service product from the drop-down menu.

.. image:: repair_orders/repair-orders-services-tab.png
   :alt: Services tab on the repair order form with a service line added.

Set the :guilabel:`Quantity` and :guilabel:`Unit` of the service, then click :guilabel:`Save`.

> 

Reviewer: ```suggestion
Set the :guilabel:`Quantity` and :guilabel:`Unit` of the service, then click :guilabel:`Save`.

.. image:: repair_orders/repair-orders-services-tab.png
   :alt: Services tab on the repair order form with a service line added.

```
I think it always looks best to have the image at the bottom - this way there's no orphan sentence after it, disconnected from the rest of the section.

Reviewer: Oh, and also make this image 933 wide

> repair order is *Available* or *Not Available* based on the availability of the parts.

Reviewer: ```suggestion
This moves the |RO| to the :guilabel:`Confirmed` stage and reserves the necessary components needed
for the repair. Once confirmed, a :guilabel:`Component Status` field appears on the |RO| form, indicating whether the
repair order is *Available* or *Not Available* based on the availability of the parts.
```

Once ready, click :guilabel:`Start Repair`. This moves the |RO| to the :guilabel:`Under Repair`

> stage. If the |RO| should be canceled instead, click :guilabel:`Cancel Repair`.

Reviewer: ```suggestion
stage. If the |RO| should be canceled instead, click :guilabel:`Cancel`.
```
I just see 'Cancel' as the button, so check this.

Once all products have been successfully repaired, the |RO| is completed. To register this in the
database, click :guilabel:`End Repair`.

.. note::
   If the quantity consumed for a part or service differs from its initial demand, clicking
   :guilabel:`End Repair` causes a :guilabel:`Consumption Warning` wizard to appear, listing the
   expected (:guilabel:`To Consume`) and actual (:guilabel:`Consumed`) quantity for each affected

>    product.

Reviewer: ```suggestion
   When the quantity consumed for a part or service differs from the initial demand, click
   :guilabel:`End Repair`, and a *Consumption Warning* pop=up window appears, listing the
   expected (:guilabel:`To Consume`) and actual (:guilabel:`Consumed`) quantity for each affected
   product.
```

>    .. image:: repair_orders/repair-orders-consumption-warning.png

Reviewer: I would move this to the bototm of the text in this block, and also crop it so there is no ne of the grey outline - you can justhave the white edges of the box visible.

      :alt: Consumption Warning wizard comparing expected and consumed quantities.

   Select one of three options to continue:

   - :guilabel:`Confirm` to validate the |RO| with the quantities as they currently are.
   - :guilabel:`Set Quantities & Validate` to update the demand of each line to match the quantity
     consumed before validating.

>    - :guilabel:`Discard` to close the wizard without ending the repair.

Reviewer: ```suggestion
   - :guilabel:`Discard` to close the pop-up window without ending the repair.
```
